src/main/java/net/snowflake/client/core/HttpUtil.java

```diff
@@ -899,4 +899,21 @@ static int convertSystemPropertyToIntValue(String systemProperty, int defaultVal
     }
     return returnVal;
   }
+
+  /**
+   * Helper function to attach additional headers to a request if present. This takes a (nullable)
+   * map of headers in <name,value> format and adds them to the incoming request using addHeader.
+   *
+   * <p>Snowsight uses this to attach headers with additional telemetry information, see
+   * https://snowflakecomputing.atlassian.net/wiki/spaces/EN/pages/2960557006/GS+Communication
+   *
+   * @param request The request to add headers to. Must not be null.
+   * @param additionalHeaders The headers to add. May be null.
+   */
+  static void applyAdditionalHeadersForSnowsight(
+      HttpRequestBase request, Map<String, String> additionalHeaders) {
+    if (additionalHeaders != null && !additionalHeaders.isEmpty()) {
+      additionalHeaders.forEach(request::addHeader);
```

Comment:
Please check input param request for null or not. If that is not possible in reality then ignore this comment. This is to protect against NPE.

Reply:
Every usage of this method is directly after the `httpRequest` variable has been instantiated with `new Get/Post` so it cannot be null.

For future usages, I have documented the javadoc of this method that `request` must be non-null.